app/models/DiagramType.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/*PLEASE DO NOT EDIT THIS CODE*/
```

Comment:
Where are the underlying ump files?  If they were borrowed from the umplifier code, I would consider that coincidental duplication and the raw sources (aka the ump files) should be included in this project.  If there are genuine similarities that are worth keeping, then we can revisit that.